### PR TITLE
Correct SPDX license identifiers

### PR DESCRIPTION
**Pull request by keithc-ca**
Issue #6995:
* "GPL-2.0" should have been "GPL-2.0-only"
* "OpenJDK-assembly-exception-1.0" replaces "Assembly-exception"

### CODE REVIEW

**Review by mstoodle — CHANGES_REQUESTED** [review]
haven't checked everything yet, but the mentioned missing characters cause build breaks

File: jitbuilder/apigen/schema/api.schema.json
Comment (keithc-ca) [reply jitbuilder/apigen/schema/api.schema.json]: Corrected in 62c1cd97089e589f0a2b440fd70929090753f153.

File: jitbuilder/apigen/schema/api-class.schema.json
Comment (mstoodle) [inline jitbuilder/apigen/schema/api-class.schema.json:7]: same missing ", at end of line

File: jitbuilder/apigen/schema/api-field.schema.json
Comment (mstoodle) [inline jitbuilder/apigen/schema/api-field.schema.json:7]: missing ",

File: jitbuilder/apigen/schema/api-service.schema.json
Comment (mstoodle) [inline jitbuilder/apigen/schema/api-service.schema.json:7]: missing ",

File: jitbuilder/apigen/schema/api-type.schema.json
Comment (mstoodle) [inline jitbuilder/apigen/schema/api-type.schema.json:7]: missing ",

**Review by mstoodle — CHANGES_REQUESTED** [review]
even json file has the same issue of missing ", at the end of the line that's been updated (except the first one I pointed out which has been fixed - thanks!)

**Comment by keithc-ca** [comment]
Hopefully I've fixed all files in 0972f4a4e2f7436e1dd51bea8156979175804c81.

**Comment by keithc-ca** [comment]
Besides the json files, there were others that didn't get (mechanically) updated properly. Marking as draft until those are corrected.

**Comment by keithc-ca** [comment]
Rebased and updated:
* fix new files added recently
* fix `*.s` files where long lines are not permitted
* correct files where the license identifier expression spanned two or more lines
* fix strange "SPDX-L$(LIBJITBUILDERDIR)cense-Identifier"

**Comment by keithc-ca** [comment]
Removed "fixes" comment; new files added by recent pull requests will need to be updated.

**Review by mstoodle — APPROVED** [review]
I found `doc/ValgrindMemcheckAPI.md` which does not have the right license string (only "SPDX-License-Identifier: EPL-2.0 OR Apache-2.0" which is, I think, our original identifier so has been missed for a long time :( ). We can handle it in this PR or under a separate one since it's a long standing mistake.

The failure on MacOS is not due to this change (another instance of #6516)

**Comment by mstoodle** [comment]
jenkins build all

**Comment by keithc-ca** [comment]
I left `doc/ValgrindMemcheckAPI.md` as it was. It didn't have any invalid identifiers discussed in the issue. If I recall correctly, there were other such files. If the license in those files should be updated, I think that should be handled separately.

**Comment by AdamBrousseau** [comment]
jenkins buld riscv

**Comment by AdamBrousseau** [comment]
jenkins build riscv

**Comment by keithc-ca** [comment]
Is this waiting for additional committer approvals, or something else?

**Pull request merged**
